src/tests/mock/http.rs

```diff
@@ -255,3 +305,155 @@ fn full_path(path: Option<String>, query: Option<String>) -> String {
         None => path,
     }
 }
+
+/// Parses dot-separated field paths into a set of paths.
+fn parse_field_paths(paths: &[&str]) -> HashSet<Vec<String>> {
+    paths
+        .iter()
+        .map(|path| path.split('.').map(String::from).collect())
+        .collect()
+}
+
+/// Recursively compares two JSON values, excluding specified paths, and returns
+/// detailed errors using anyhow.
+fn json_matches_excluding(
```

Comment:
This seems quite complex. Wouldn't it be easier to do the following?
1. clone the actual value
2. delete the ignored paths from it
3. use strict equality again

That way we don't need recursion and the code would probably be self explanatory.

Reply:
I was a bit tired of seeing "right doesn't equal left" which requires manually finding the differences, which is annoying.

Reply:
I think giving better error messages is a good reason. But if this is then intended as a more versatile building block I think we probably should put it into `shared` so we can also make use of it in the backend repo while not duplicating code. 🤔

Reply:
https://github.com/cowprotocol/services/pull/2783